src_py/sprite.py

```diff
@@ -1099,17 +1099,14 @@ def draw(self, surface, bgd=None):
                     if spr._visible:
                         # sprite not dirty; blit only the intersecting part
                         _spr_rect = spr.rect
-                        if spr.source_rect is not None:
-                            _spr_rect = Rect(spr.rect.topleft,
-                                             spr.source_rect.size)
                         _spr_rect_clip = _spr_rect.clip
                         for idx in _spr_rect.collidelistall(_update):
                             # clip
                             clip = _spr_rect_clip(_update[idx])
                             _surf_blit(spr.image,
                                        clip,
-                                       (clip[0] - _spr_rect[0],
-                                        clip[1] - _spr_rect[1],
+                                       (clip[0] - _spr_rect[0] + spr.source_rect[0],
```

Comment:
The default value for `source_rect` is `None`, so this will crash when a `source_rect` is not used.

Reply:
Thanks for the review and feedback.

You are right, the source_rect can be None. I have considered it now in the second commit.

About the different sizes... it does not make much sense because its the source area of the blit method (not sure what will happen if you have a bigger source rect...... for a smaller source rect I guess you only the that part).

I still wonder if the clipping will work in all cases.